src/Components/Locations.js

```diff
@@ -0,0 +1,54 @@
+import { useState, useEffect } from 'react';
+import axios from "axios";
+
+export default function Locations() {
+    const [list, setList] = useState([]);
+    const [showList, setShowList] = useState(false)
+    useEffect(() => {
+        generateList()
+    }, [])
+    const generateList = async () => {
+        try {
+            const res = await axios.get(
+                `https://ghibliapi.herokuapp.com/locations`
+            )
+            setList( res.data)
+        } catch (err) {
+            setList( "")
+        }
+    }
+    const handleClick = () => {
+        setShowList((prev) => !prev)
+    }
+    let listOfLocations = []
+    if (list.length) {
+        listOfLocations = list.map((location, i) =>
+            <li key={i}>
+                Name: {location.name}<br/>
+                Climate: {location.climate}<br/>
+                Terrain: {location.terrain}<br/>
+                <br/>
+            </li>
+        )
+    }
+
+    if (showList === false) {
```

Comment:
I'd prefer to see these types of conditions switched and looking for truthyness. So if(showList) { we show the list. 
Will read a little better.